src/JSInterop/Microsoft.JSInterop/src/JSRuntime.cs

```diff
@@ -178,15 +184,45 @@ protected virtual void BeginInvokeJS(long taskId, string identifier, string? arg
             in DotNetInvocationResult invocationResult);
 
         /// <summary>
-        /// Invoked by JS 
+        /// Invoked by JS
         /// </summary>
         /// <param name="id">Atomically incrementing identifier for the byte array being transfered.</param>
         /// <param name="data">Byte array to be transfered to JS.</param>
-        protected internal virtual void SupplyByteArray(long id, byte[] data)
+        protected internal virtual void SendByteArray(int id, byte[] data)
         {
             throw new NotSupportedException("JSRuntime subclasses are responsible for implementing byte array transfer to JS.");
         }
 
+        /// <summary>
+        /// Accepts the byte array data being transferred from JS to DotNet.
+        /// </summary>
+        /// <param name="id">Identifier for the byte array being transfered.</param>
+        /// <param name="data">Byte array to be transfered from JS.</param>
+        public void ReceiveByteArray(int id, byte[] data)
```

Comment:
Ah, now I remember why we direct the calls through `DotNetDispatcher`. It's to do with access modifiers.

Really sorry we sent you in a misleading direction with an earlier comment. You were correct to have `ReceiveByteArray` initially as a `public static` method on `DotNetDispatcher`. That should then call into this method here on `JSRuntime` (which should be `internal`) to actually perform the work. This is to avoid confusing customers by having a public method on `JSRuntime` (which is a type they can legitimately work with) that they are not meant to call.

Hopefully this is only a small tweak now! Apologies for not realising this earlier. I knew there was some reason for the existing pattern but lost track of what it was.

Reply:
Good catch

Reply:
> That should then call into this method here on JSRuntime (which should be internal) 

I believe it'll have to be `protected internal` to ensure the (now) virtual method can be overridden in the `RemoteJSRuntime`. Please do let me know if you have any concerns with that 😄